impl/client/src/main/javascript/web/pentaho/data/filter/_core/isLike.js

```diff
@@ -82,7 +83,13 @@ define([
             return false;
           }
 
-          var firstOccurrence = formatted.indexOf(referenceFormatted);
+          var firstOccurrence;
+
+          if(isCaseInsensitive) {
+            firstOccurrence = formatted.toLowerCase().indexOf(referenceFormatted.toLowerCase());
```

Comment:
Please perform the `referenceFormatted.toLowerCase()` once, just after the declaration of the `referenceFormatted` variable.